patches/systemd-utils/0001-Add-managarm-support.patch

```diff
@@ -0,0 +1,906 @@
+                 unsigned flags) {
+ 
+-#  if defined __NR_move_mount && __NR_move_mount >= 0
++#  if defined __NR_move_mount && __NR_move_mount >= 0 && !defined(__managarm__)
+         return syscall(__NR_move_mount, from_dfd, from_pathname, to_dfd, to_pathname, flags);
+ #  else
+         errno = ENOSYS;
+@@ -587,7 +605,7 @@ static inline int missing_move_mount(
+ #endif
+ 
+ static inline int missing_fsopen(const char *fsname, unsigned flags) {
+-#  if defined __NR_fsopen && __NR_fsopen >= 0
++#  if defined __NR_fsopen && __NR_fsopen >= 0 && !defined(__managarm__)
+         return syscall(__NR_fsopen, fsname, flags);
+ #  else
+         errno = ENOSYS;
+@@ -619,7 +637,7 @@ static inline int missing_fsopen(const char *fsname, unsigned flags) {
+ #endif
+ 
+ static inline int missing_fsconfig(int fd, unsigned cmd, const char *key, const void *value, int aux) {
+-#  if defined __NR_fsconfig && __NR_fsconfig >= 0
++#  if defined __NR_fsconfig && __NR_fsconfig >= 0 && !defined(__managarm__)
+         return syscall(__NR_fsconfig, fd, cmd, key, value, aux);
+ #  else
+         errno = ENOSYS;
+@@ -639,7 +657,7 @@ static inline int missing_fsconfig(int fd, unsigned cmd, const char *key, const
+ #endif
+ 
+ static inline int missing_fsmount(int fd, unsigned flags, unsigned ms_flags) {
+-#  if defined __NR_fsmount && __NR_fsmount >= 0
++#  if defined __NR_fsmount && __NR_fsmount >= 0 && !defined(__managarm__)
+         return syscall(__NR_fsmount, fd, flags, ms_flags);
+ #  else
+         errno = ENOSYS;
+@@ -655,7 +673,7 @@ static inline int missing_fsmount(int fd, unsigned flags, unsigned ms_flags) {
+ #if !HAVE_GETDENTS64
+ 
+ static inline ssize_t missing_getdents64(int fd, void *buffer, size_t length) {
+-#  if defined __NR_getdents64 && __NR_getdents64 >= 0
++#  if defined __NR_getdents64 && __NR_getdents64 >= 0 && !defined(__managarm__)
+         return syscall(__NR_getdents64, fd, buffer, length);
+ #  else
+         errno = ENOSYS;
+diff --git a/src/basic/mountpoint-util.c b/src/basic/mountpoint-util.c
+index 135b296..9216102 100644
+--- a/src/basic/mountpoint-util.c
++++ b/src/basic/mountpoint-util.c
+@@ -554,6 +554,10 @@ int dev_is_devtmpfs(void) {
+         int mount_id, r;
+         char *e;
+ 
++#if defined(__managarm__)
++        return true;
++#endif
++
+         r = path_get_mnt_id("/dev", &mount_id);
+         if (r < 0)
+                 return r;
+diff --git a/src/basic/process-util.c b/src/basic/process-util.c
+index d9e8bc9..e90c367 100644
+--- a/src/basic/process-util.c
++++ b/src/basic/process-util.c
+@@ -1582,7 +1582,7 @@ int safe_fork_full(
+         if (flags & (FORK_DEATHSIG_SIGTERM|FORK_DEATHSIG_SIGINT|FORK_DEATHSIG_SIGKILL))
+                 if (prctl(PR_SET_PDEATHSIG, fork_flags_to_signal(flags)) < 0) {
+                         log_full_errno(prio, errno, "Failed to set death signal: %m");
+-                        _exit(EXIT_FAILURE);
++                        // _exit(EXIT_FAILURE);
```

Comment:
Please add a comment to justify this.

Reply:
Ideally we should implement this, or at least indeed slap a big TODO on it.